src/main/webapp/inc.modul.jsp

```diff
@@ -1,1772 +1,1470 @@
-				try {
-					if (null != cn)
-						cn.close();
-				} catch (Exception ex) {
-				}
-			}
-			out.println("</table>\n");
-		}
-	}
-
-	else if (formular.equals("person")) {
-		if (modul.equals("namen")) {
-			out.println("<tr><td>\n");
-
-	Connection cn = null;
-			Statement st = null;
-			ResultSet rs = null;
-			try {
-				Class.forName(sqlDriver);
-				cn = DriverManager.getConnection(sqlURL, sqlUser,
-						sqlPassword);
-				st = cn.createStatement();
-
-				rs = st
-						.executeQuery("SELECT count(DISTINCT namenkommentar.ID) FROM einzelbeleg LEFT OUTER JOIN "+
-						"einzelbeleg_hatperson ON "+
-						"einzelbeleg.ID=einzelbeleg_hatperson.EinzelbelegID LEFT OUTER JOIN "+
-						"person ON einzelbeleg_hatperson.PersonID=person.ID LEFT OUTER JOIN "+
-						"einzelbeleg_hatnamenkommentar ON "+
-						"einzelbeleg_hatnamenkommentar.EinzelbelegID=einzelbeleg.ID "+
-						"LEFT OUTER JOIN namenkommentar ON "+
-						"namenkommentar.ID=einzelbeleg_hatnamenkommentar.NamenkommentarID WHERE person.ID=\""
-								+ id + "\"");
-				out.print("<label>Name");
-				if(rs.next() && rs.getInt(1)>1) out.println("n");
-				out.println("</label></td><td>");
-				rs = st
-						.executeQuery("SELECT DISTINCT namenkommentar.PLemma, "+
-						"namenkommentar.ID FROM einzelbeleg LEFT OUTER JOIN "+
-						"einzelbeleg_hatperson ON "+
-						"einzelbeleg.ID=einzelbeleg_hatperson.EinzelbelegID LEFT OUTER JOIN "+
-						"person ON einzelbeleg_hatperson.PersonID=person.ID LEFT OUTER JOIN "+
-						"einzelbeleg_hatnamenkommentar ON "+
-						"einzelbeleg_hatnamenkommentar.EinzelbelegID=einzelbeleg.ID "+
-						"LEFT OUTER JOIN namenkommentar ON "+
-						"namenkommentar.ID=einzelbeleg_hatnamenkommentar.NamenkommentarID WHERE person.ID=\""
-								+ id + "\"");
-				while (rs.next()  && rs.getString("namenkommentar.PLemma") !=null) {
-					out.println("<a href=\"namenkommentar?ID="+rs.getString("namenkommentar.ID")+"\">"+format(rs.getString("namenkommentar.PLemma"),"PLemma")+"<br>");
-				}
-				out.println("</td></tr>");
-			} catch (Exception e) {
-				out.println(e);
-			} finally {
-				try {
-					if (null != rs)
-						rs.close();
-				} catch (Exception ex) {
-				}
-				try {
-					if (null != st)
-						st.close();
-				} catch (Exception ex) {
-				}
-				try {
-					if (null != cn)
-						cn.close();
-				} catch (Exception ex) {
-				}
-			}
-			out.println("</ul>\n");
-		}if (modul.equals("nachweise")) {
-			out.println("<table>\n");
+            try {
+
+                List<Object[]> resultList = ModulIncDB.getListEinzelbelegTextkritik(id);
+
+                int count = 0;
+                for (Object[] row : resultList) {
+                    count++;
+                    if (count % 2 == 0) {
+                        out.println("<tr>");
+                    } else {
+                        out.println("<tr bgcolor='#AACCDD'>");
+                    }
+
+                    String zitierweise = row[0] != null ? String.valueOf(row[0]) : "";
```

Comment:
Before this PR, it was possible to access columns in the ResultSet via the proper column name (e.g. "ed.Zitierweise"), but now they are accessed via a counter. The queries used in the ModulIncDB are very complex with tons of different columns. If the query is ever changed, the counter will also change which is very dangerous. Can we find any solution to keep the addressing via column name here?

Reply:
As discussed => no, this is related to the columns inside the sql result row. It seems to be a lot of effort to correct this now, but since it is only used as single select query in the module it might be locally enough to keep it like this. However, i'd strongly prefer to keep the by-string accessor for everything else which is coded in the future.